package/src/components/Message/hooks/useProcessReactions.ts

```diff
@@ -23,12 +22,20 @@ export type ReactionSummary = {
 
 export type ReactionsComparator = (a: ReactionSummary, b: ReactionSummary) => number;
 
+export type MessageReactionsData<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+> = {
+  /** An array of the reaction objects to display in the list */
+  latest_reactions?: ReactionResponse<StreamChatGenerics>[];
```

Comment:
Although this is cleaner so to speak, something I don't particularly like about it is the fact that it has the potential to break our contract here. Let's hypothetically say that at one point `latest_reactions` gets its type changed from `ReactionResponse<SCG>[]` to `string[]` (keeping only reaction IDs for example). Typescript will not complain that something broke, since we explicitly define these here (more importantly, it won't reflect the change wherever we want to use this).

Unless we really explicitly want to pin these as defined, I would keep the old approach for now.

Reply:
You mean keep it in ReactionList component?

Reply:
I have changed it to:

```tsx
import { MessageResponse, ReactionResponse } from 'stream-chat';

export type MessageReactionsData<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
> = Pick<
  MessageResponse<StreamChatGenerics>,
  'own_reactions' | 'latest_reactions' | 'reaction_groups'
>;
```

I think this will fix it once and for all. Thanks for pointing this out. 👍🏻